### ISSUE TITLE
Hard to get a good performance

### ISSUE THREAD

**Issue by iced-cappuccino** [jesbu1/hidio#10]
Hi, did you use any practical trick during training procedure? And was the feature extractor trained through supervised learning? Since it is hard for me to get a good performance when I train the hidio model from scratch. 

Also, I'm confused about the code shown in your pseudo code list on page 13. Why is it "Compute intrinsic rewards $r_{h,k}^{lo}=q_{\psi}(u_{h}|\bar{a}_{h,k}\bar{s}_{h,k+1})$" rather than "Compute intrinsic rewards $r_{h,k}^{lo}=\logq_{\psi}(u_{h}|\bar{a}_{h,k}\bar{s}_{h,k+1})$"?

Thanks for your help!

**Comment by hnyu** [comment]
> Hi, did you use any practical trick during training procedure? And was the feature extractor trained through supervised learning? Since it is hard for me to get a good performance when I train the hidio model from scratch.
> 
> Thanks for your help!

Hi @iced-cappuccino , the feature extractor was trained together with the entire method, as the paper describes. 

Do you mean it's hard to reproduce the reported performance using the provided code, or apply HIDIO to other tasks using your own implementation? 

(Implementation details are included in: https://github.com/jesbu1/hidio/tree/master/hidio/algorithm. Also you can check out the configuration details in the example gin files.)

**Comment by iced-cappuccino** [comment]
Thanks for your reply!

Yes, I am applying HIDIO to other tasks using my own implementation.

I'm confused about the code shown in your pseudo code list on page 13. Why is it "Compute intrinsic rewards $r_{h,k}^{lo} = q_{\psi} (u_{h} | \bar{a}_{h,k}, \bar{s}_{h,k+1}) $" rather than "Compute intrinsic rewards  $ r_{h,k}^{lo}= \log q_{\psi} (u_{h} | \bar{a}_{h,k}, \bar{s}_{h,k+1}) $ "?

**Comment by hnyu** [comment]
> Thanks for your reply!
> 
> Yes, I am applying HIDIO to other tasks using my own implementation.
> 
> I'm confused about the code shown in your pseudo code list on page 13. Why is it "Compute intrinsic rewards $r_{h,k}^{lo} = q_{\psi} (u_{h} | \bar{a}_{h,k}, \bar{s}_{h,k+1}) $" rather than "Compute intrinsic rewards $ r_{h,k}^{lo}= \log q_{\psi} (u_{h} | \bar{a}_{h,k}, \bar{s}_{h,k+1}) $ "?

That should be a typo. It's log probability in fact (section 3.3). Thanks for catching this. (It might be possible to directly use the probability as the intrinsic reward, however. Or doing some other transform on the probability.)

Yeah if you apply HIDIO to other tasks, it might take some efforts to tune the hyperparameters. We have listed the important ones in the appendix D. I suggest first trying to get your implementation working on the tasks in the paper, to get a feeling of hyperparameter tuning.

**Comment by iced-cappuccino** [comment]
Yes, got it.

Thanks for your help!
